src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -998,30 +998,36 @@ type CachingStrategy =
 
 [<NoComparison; NoEquality; RequireQualifiedAccess>]
 type AccessStrategy<'event,'state> =
-    /// Allow events that pass the `isOrigin` test to be used in lieu of folding all the events from the start of the stream
-    /// When saving, `unfold` the 'state, representing it as event records stored in the Tip
-    | Unfolded of isOrigin: ('event -> bool) * unfold: ('state -> 'event seq)
+    /// Don't apply any optimized reading logic strategy. Note this can be cost prohibitive
+    /// and severely impact system scalability and hence should only be used with careful consideration.
+    | Unoptimized
     /// Simplified version of Unfolded that only saves a single Event Type
     /// Provides equivalent performance to Unfolded, just simplified function signatures
-    | Snapshot of isValid: ('event -> bool) * generate: ('state -> 'event)
-    /// Treat every known event type in the Union as being an Origin event
-    | AnyKnownEventType
+    | Snapshot of isOrigin: ('event -> bool) * toSnapshot: ('state -> 'event)
+    /// Instead of actually storing the events representing the decisions, only ever update the snapshot stored in the tip
+    | RollingState of toSnapshot: ('state -> 'event)
+    /// Allow events that pass the `isOrigin` test to be used in lieu of folding all the events from the start of the stream
+    /// When saving, uses `toEvents` to 'unfold' the 'state, representing it as event records to be stored in the Tip with efficient read cost
+    | MultiSnapshot of isOrigin: ('event -> bool) * toSnapshots: ('state -> 'event seq)
     /// Allow produced events to be transmuted to unfolds.
     /// In this mode, Optimistic Concurrency Control OCC is based on the _etag (rather than the normal Expected Version strategy)
-    | RollingUnfolds of isOrigin: ('event -> bool) * transmute: ('event list -> 'state -> 'event list*'event list)
+    | Custom of isOrigin: ('event -> bool) * transmute: ('event list -> 'state -> 'event list*'event list)
+    /// Treat every known event type in the Union as being an Origin event
+    | LatestKnownEvent
```

Comment:
xmldoc
```
/// Treat every event that we can decode as an Origin event, i.e. only ever load one event and build state from that, implying we never need to reference any events aside from the latest to know the state
/// <remarks>
Used where every event represented on the stream is sufficient to build a state from
/// A copy of the event is also retained in the `Tip` document in order that the state of the stream can be retrieved using a single (cached, etag-checked) point read.
/// </remarks>
```

the reference to the Tip document line should not go on the ES or SSS versions but the rest should be the same